core/merge.js

```diff
@@ -331,26 +331,25 @@ class Merge {
 
   // Move a folder and all the things inside it
   async moveFolderRecursivelyAsync (side /*: SideName */, folder /*: Metadata */, was /*: Metadata */) {
-    let docs
-    try {
-      docs = await this.pouch.byRecursivePathAsync(was._id)
-    } catch (err) {
-      throw err
-    }
+    const docs = await this.pouch.byRecursivePathAsync(was._id)
     move(was, folder)
     let bulk = [was, folder]
-    for (let doc of Array.from(docs)) {
+
+    const modifiedIDs = docs.map((doc) => doc._id.replace(was._id, folder._id))
+    const result = await this.pouch.db.allDocs({keys: modifiedIDs})
+    const existingsDestinations = {}
+    for (let r of result.rows) if (r.value) existingsDestinations[r.key] = r.value.rev
```

Comment:
Maybe rename const, extract revs retrieval to Pouch and prevent PLOP? e.g.

```js
// Pouch

async allRevsByIds (ids /*: string */) /*: Promise<Metadata[]> */ {
  const results = await this.db.allDocs({keys: ids})
  return _.chain(results).filter('value').map(({value}) => [value.id, value.rev]).fromPairs().value()
}

// Merge
const existingDestinationRevs = await this.pouch.allRevsByIds(destinationIds)
```

(this is textarea driven development, commit & deploy without fear :sunglasses:)

Reply:
My bad, return type should be `Promise<{[string]: string}>`

Reply:
Bonus point if we alias `type Rev = string` & `type Id = string` for documentation purpose :rainbow: 
So we could write `Promise<{[Id]: Rev}>`